### ISSUE TITLE
Book outline is available for other content types

### ISSUE THREAD

**Issue by finnlewis** [localgovdrupal/localgov_publications#83]
When creating a service page, I can add this to a book outline.

We think this is not desirable.

![Image](https://github.com/localgovdrupal/localgov_publications/assets/326588/e9bc5769-2587-40f0-8733-b7c913ecf401)

**Comment by andybroomfield** [comment]
~~can you check with the editor only permission. I think admins can add any page to a book.~~

Looks like the permission is avalible to all, might need a split permission and a form_alter hook

**Comment by rupertj** [comment]
I agree that this isn't desirable, but also, part of the plan for this module was to not interfere with how book module works, in case people want to use it.

Isn't this field being present on all the content types how book works? We could remove that field on the other content types, but we'd be deviating from the plan of not interfering with book.

I wonder if having a setting along the lines of "Let me use book for other things too please" would be a good idea? Then we could hide things book does from people who only want to use publications, and still let people who want to use book, use it how they did before they installed publications.

**Comment by rupertj** [comment]
![image](https://github.com/localgovdrupal/localgov_publications/assets/326243/132020d0-47d6-40f9-83c9-22f8c3491c03)

Lizzi from Bracknell posted this in slack today, with the question "If I create a publication cover, then pages within that publication, how do we then link to the first page of the publication as wouldn't that then be linking to the cover page."

I think she's ended up where she is from putting the cover page in the publication (possibly even creating the publication from the cover page) and then linking to a page in the publication with the reference field.

I think hiding the Publication/Book fieldset from all content types bar Publication Page, with a setting to stop that happening in case people want to use Book would make the UI more intuitive.

**Comment by rupertj** [comment]
So this happens because we grant the "administer book outlines" permission to Editors by default. That permission also lets you add any content type to a book. Users without that permission can only add certain content types (Set at /admin/structure/book/settings) to books.

So we should:
1) Stop granting that permission and either a) communicate to site owners that they should remove the permission or b) remove it in an update hook.

2) Change anywhere we're using that permission for our own devices. It's used in the routing file to allow access to the publications listing at least.

**Comment by AWearring** [comment]
@rupertj @andybroomfield 
I found that removing the "administer book outlines" permission caused errors/warnings on the Publication Cover page Add/Edit forms:
![image](https://github.com/localgovdrupal/localgov_publications/assets/95761929/fa1f385c-ca05-484e-9110-3fa027a7f54b)

Removing 
```
'node_localgov_publication_cover_page_form',
'node_localgov_publication_cover_page_edit_form',
```
from `$forms_to_alter` in the `localgov_publications_form_alter()` function in `localgov_publications.module` resolves this.

Not sure if that adversely effects anything, but this solves the issue raised by @finnlewis

**Comment by msayoung** [comment]
+1 to @AWearring 's solution

Is there likely to be any problems removing those lines?

**Comment by rupertj** [comment]
I've taken out the lines mentioned while I was in there working on #218. Once that's merged we can take a look at the permissions and try again.

**Comment by markconroy** [comment]
I'm getting the same error when logged in as a **news editor** and trying to change a news article from 'published' to 'draft'. 

If I edit a news article from published and leave it published, it's fine. But if I edit a news article and set it to draft, I get this:

```

Error message

    Warning: Trying to access array offset on null in Drupal\book\Plugin\Validation\Constraint\BookOutlineConstraintValidator->validate() (line 56 of modules/contrib/book/src/Plugin/Validation/Constraint/BookOutlineConstraintValidator.php).

    Warning: Trying to access array offset on null in Drupal\book\Plugin\Validation\Constraint\BookOutlineConstraintValidator->validate() (line 62 of modules/contrib/book/src/Plugin/Validation/Constraint/BookOutlineConstraintValidator.php).

    Warning: Trying to access array offset on null in Drupal\book\Plugin\Validation\Constraint\BookOutlineConstraintValidator->validate() (line 68 of modules/contrib/book/src/Plugin/Validation/Constraint/BookOutlineConstraintValidator.php).

    You can only change the book outline for the published version of this content.

```

I don't think we should need to give News Editor role the 'Administer book outline' permission.

**Comment by rupertj** [comment]
@markconroy That's a different issue. I reckon the cause is this https://www.drupal.org/project/book/issues/2918537 and we could fix it by upgrading to book 2, which there's an existing issue for here: https://github.com/localgovdrupal/localgov_publications/issues/222

**Comment by markconroy** [comment]
Thanks for that update @rupertj 

Hopefully we'll start making progress on moving to Book 2, would be great to have this sorted.

**Comment by tonypaulbarker** [comment]
Have closed the Github issue https://github.com/localgovdrupal/localgov_publications/issues/222 for allowing 2.x of book and opened on drupal.org https://www.drupal.org/project/localgov_publications/issues/3563646
